promql/functions.go

```diff
@@ -60,10 +61,15 @@ func extrapolatedRate(vals []parser.Value, args parser.Expressions, enh *EvalNod
 	ms := args[0].(*parser.MatrixSelector)
 	vs := ms.VectorSelector.(*parser.VectorSelector)
 
+	enhTs := enh.Ts
+	if vs.Timestamp > 0 {
```

Comment:
Since this method touches the timestamp of the samples, I think we need to adjust it here. Because in case of matrix selector or subquery, the samples will have the timestamp relative to the timestamp passed with @.

Reply:
That doesn't sound right, nothing in this file should care about this feature. It's a matter of passing in the right data.